File: src/client/java/net/empyrean/config/ClientConfig.java
package net.empyrean.config;

import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;

@Config(name = "empyrean")
public class ClientConfig implements ConfigData {
    public boolean doNotTouchEnablesSinglePlayer = false;
}

File: src/client/kotlin/net/empyrean/config/ClientConfig.kt
package net.empyrean.config

import net.empyrean.config.impl.ConfigProvider

object ClientConfig : ConfigProvider() {
    @JvmStatic
    val compactEffectDisplay: Boolean by cfg()

    @JvmStatic
    val displayEffectLength: Boolean by cfg()

    @JvmStatic
    val enableSingleplayer: Boolean by cfg()

    override fun init() {
        category("stable") {
            group("effects") {
                boolean("compactEffectDisplay", true)
                boolean("displayEffectLength", true)
            }
        }
        category("experimental") {
            group("expSingleplayer") {
                boolean("enableSingleplayer", false)
            }
        }
        saveDefault()
        load()
    }

}